src/lib/support/PersistedCounter.h

```diff
@@ -110,6 +110,45 @@ class PersistedCounter : public MonotonicallyIncreasingCounter<T>
         return MonotonicallyIncreasingCounter<T>::Init(startValue);
     }
 
+    CHIP_ERROR SetValue(T value) override
+    {
+        VerifyOrReturnError(mStorage != nullptr, CHIP_ERROR_INCORRECT_STATE);
+        VerifyOrReturnError(mKey.IsInitialized(), CHIP_ERROR_INCORRECT_STATE);
+
+        // Calculate previous Epoch from value
+        T previousEpoch = value - (value % mEpoch);
+
+#if CHIP_CONFIG_PERSISTED_COUNTER_DEBUG_LOGGING
+        if constexpr (std::is_same_v<decltype(previousEpoch), uint64_t>)
+        {
+            ChipLogDetail(EventLogging,
+                          "PersistedCounter::SetValue() newValue 0x" ChipLogFormatX64 " previousEpoch 0x" ChipLogFormatX64,
+                          ChipLogValueX64(value), ChipLogValueX64(previousEpoch));
+        }
+        else if (std::is_same_v<decltype(previousEpoch), uint32_t>)
+        {
+            ChipLogDetail(EventLogging, "PersistedCounter::SetValue() newValue 0x%" PRIx32 " previousEpoch 0x%" PRIx32,
+                          static_cast<uint32_t>(value), static_cast<uint32_t>(previousEpoch));
+        }
+#endif
+
+        // Set next Epoch value
+        ReturnErrorOnFailure(PersistNextEpochStart(previousEpoch + mEpoch));
+
+        // Set new mCounterValue
+        ReturnErrorOnFailure(MonotonicallyIncreasingCounter<T>::SetValue(value));
+
+        // Advancing the epoch should have ensured that the current value is valid
+        VerifyOrReturnError(MonotonicallyIncreasingCounter<T>::GetValue() + mEpoch >= mNextEpoch, CHIP_ERROR_INTERNAL);
```

Comment:
This has the same rollover problems as the commented-out checks here, no?  That is, if we are in the `>` case, not `==`, then we could test false if `mNextEpoch` is the mac counter value....

It seems like if these counters can roll over we need to be _very_ clear about what our `>` and `<` mean, or just use equality checks....